Source/path.cpp

```diff
@@ -30,8 +30,18 @@ PATHNODE *pnode_tblptr[MAXPATHNODES];
 PATHNODE *path_2_nodes;
 
 /** For iterating over the 8 possible movement directions */
-const char pathxdir[8] = { -1, -1, 1, 1, -1, 0, 1, 0 };
-const char pathydir[8] = { -1, 1, -1, 1, 0, -1, 0, 1 };
+const Displacement PathDirs[8] = {
+	// clang-format off
+	{ -1, -1 },
+	{ -1,  1 },
+	{  1, -1 },
+	{  1,  1 },
+	{ -1,  0 },
+	{  0, -1 },
+	{  1,  0 },
+	{  0,  1 },
+	// clang-format on
+};

```

Comment:
```suggestion
const Direction PathDirs[8] = {
	// clang-format off
	Direction::DIR_N,
	Direction::DIR_W,
	Direction::DIR_E,
	Direction::DIR_S,
	Direction::DIR_NW,
	Direction::DIR_NE,
	Direction::DIR_SE,
	Direction::DIR_SW,
	// clang-format on
};
```
This is an array of directions, it can use the Direction enum values.

Reply:
I went with this to avoid the issue in your next comment :)

Since you approved the PR and it was set to auto-merge it's already been merged, but you are welcome to make the change if you prefer.

Reply:
Wasn't worth holding up this PR over, and it's easier to implement in isolation anyway :). I'll leave it for #2260 or a followup story